frontend/services/entities/software.ts

```diff
@@ -234,7 +234,39 @@ export default {
       true
     );
   },
+  // TODO: Only send data that is changing and test E2E
+  editSoftwarePackage: (
+    softwareId: number,
+    data: IAddPackageFormData,
+    timeout?: number
+  ) => {
+    const { SOFTWARE_TITLE } = endpoints;
+
+    if (!data.software) {
+      throw new Error("Software package is required");
+    }
 
+    const formData = new FormData();
+    formData.append("software", data.software);
+    formData.append("self_service", data.selfService.toString());
+    data.installScript && formData.append("install_script", data.installScript);
```

Comment:
Since the UI should populate these from the existing installer, we don't have to conditionally include them. Blanking them out will set back to defaults, which is what we want to install/post-install/uninstall scripts (pre-install query is blank by default, and we want to pass those in).

So the only cases where we might not send a field would be the installer itself I think. Sorry for the confusion here; the endpoint will include the extra optionality but for the frontend it doesn't make sense to omit fields, with the above exceptions.

Reply:
^ Forgot to tag @RachelElysia